dashboard/src/components/branding/che-branding.factory.ts

```diff
@@ -130,6 +134,7 @@ export class CheBranding {
   updateData(): void {
     this.$http.get(ASSET_PREFIX + 'product.json').then(res => res.data).then((branding: IBranding) => {
       this.branding = branding;
+      this.initialize();
```

Comment:
@akurinnoy  Why should we initialize more than once?

Whan about 
```
if (!this.$rootScope.branding) {
  this.initialize();
}
```

Reply:
thanks, fixed

Reply:
And one more - we can initialize only once:
```
  initialize(): void {
    Object.defineProperty(this.$rootScope, 'branding', {
      get: () => {
        return {
          title: this.getProductName(),
          name: this.getName(),
          logoURL: this.getProductLogo(),
          logoText: this.getProductLogoText(),
          favicon: this.getProductFavicon(),
          loaderURL: this.getLoaderUrl(),
          websocketContext: this.getWebsocketContext(),
          helpPath: this.getProductHelpPath(),
          helpTitle: this.getProductHelpTitle(),
          footer: this.getFooter(),
          supportEmail: this.getProductSupportEmail(),
          oauthDocs: this.getOauthDocs(),
          cli: this.getCLI(),
          docs: this.getDocs(),
          workspace: this.getWorkspace()
        };
      }
    });
  }
```
WDYT?

Reply:
I prefer not to do that right now because of CI doesn't work stably today and it will take additional time to get tests passed or manually check failed ones.
But you can file an issue about this improvement.